examples/kotlin-matter-controller/java/src/com/matter/controller/commands/common/Argument.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package com.matter.controller.commands.common
+
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicLong
+
+class Argument {
```

Comment:
please add a TODO to re-make this class. This seems to be a mix of several classes into one.

Instead, some sealed class/interface and actual implementations for each would make a lot more sense (e.g. min/max value has no meaning for IPAddress or strings, value being Any seems wrong etc.).

Reply:
https://github.com/project-chip/connectedhomeip/issues/29632